src/main/resources/typescriptc.js

```diff
@@ -266,6 +294,17 @@
             };
             output.results.push(result);
         });
+
+        // If we are outputting a single file create the array of written files
+        //  and map it against all input files
+        if (options.out) {
+            var filesWritten = createFilesWrittenArray(options.out);
```

Comment:
The reason for this is that the `createFilesWrittennArray` function calls `fixSourceMapFile`, which is an expensive operation; as it reads the whole source map file in, parses the JSON, maps a small change and then writes it out again.

When operating with the single outFile it is extremely costly to run on a large project (my test project has over 500 source files for instance) as the map file ends up being quite large. This small change saved ~90% of the time for me.